src/devices/bus/epson_qx/multifont.cpp

```diff
@@ -0,0 +1,284 @@
+	device_t(mconfig, EPSON_QX_OPTION_MULTIFONT, tag, owner, clock),
+	device_option_expansion_interface(mconfig, *this),
+	m_mcu(*this, "mcu"),
+	m_i8243(*this, "i8243"),
+	m_fonts(*this, "font%u",0),
+	m_ioport(*this, "IOBASE")
+{
+}
+
+//-------------------------------------------------
+//  device_input_ports - device-specific ports
+//-------------------------------------------------
+ioport_constructor multifont_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME(multifont);
+}
+
+//-------------------------------------------------
+//  device_rom_region - device-specific roms
+//-------------------------------------------------
+const tiny_rom_entry *multifont_device::device_rom_region() const
+{
+	return ROM_NAME(multifont);
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - device-specific config
+//-------------------------------------------------
+void multifont_device::device_add_mconfig(machine_config &config)
+{
+	I8039(config, m_mcu, 9_MHz_XTAL);
+	m_mcu->set_addrmap(AS_PROGRAM, &multifont_device::rom_map);
+	m_mcu->prog_out_cb().set(m_i8243, FUNC(i8243_device::prog_w));
+	m_mcu->p1_out_cb().set(FUNC(multifont_device::p1_w));
+	m_mcu->p2_out_cb().set(FUNC(multifont_device::p2_w));
+	m_mcu->bus_out_cb().set(FUNC(multifont_device::write_bus));
+	m_mcu->bus_in_cb().set(FUNC(multifont_device::read_bus));
+	m_mcu->t1_in_cb().set(FUNC(multifont_device::t1_r));
+	m_mcu->t0_in_cb().set(FUNC(multifont_device::t0_r));
+	I8243(config, m_i8243);
+	m_i8243->p4_out_cb().set(FUNC(multifont_device::rom_addr_w<0>));
+	m_i8243->p5_out_cb().set(FUNC(multifont_device::rom_addr_w<4>));
+	m_i8243->p6_out_cb().set(FUNC(multifont_device::rom_addr_w<8>));
+	m_i8243->p7_out_cb().set(FUNC(multifont_device::rom_select_w));
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+void multifont_device::device_start()
+{
+	m_iobase = 0x00;
+	m_data_out = 0x00;
+	m_data_in = 0x00;
+	m_p1 = 0xff;
+	m_p2 = 0xff;
+
+	save_item(NAME(m_iobase));
+	save_item(NAME(m_status));
+	save_item(NAME(m_data_out));
+	save_item(NAME(m_data_in));
+	save_item(NAME(m_p1));
+	save_item(NAME(m_p2));
+	save_item(NAME(m_rom_sel));
+	save_item(NAME(m_address));
+	save_item(NAME(m_rom_bank));
+	save_item(NAME(m_bus_reset));
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+void multifont_device::device_reset()
+{
+	m_status = 0x00;
+	address_space &space = m_bus->iospace();
+
+	if (m_iobase != 0) {
+		space.unmap_readwrite(m_iobase, m_iobase+1);
+	}
+	m_iobase = (m_ioport->read() & 0x06) + 0xf8;
+
+	space.install_device(m_iobase, m_iobase+1, *this, &multifont_device::map);
+
+	m_bus_reset = true;
+}
+
+uint8_t multifont_device::t0_r()
+{
+	return m_status & 0x01;
+}
+
+uint8_t multifont_device::t1_r()
+{
+	return (m_status & 0x80) >> 7;
```

Comment:
You can use the `BIT` helper for extracting a bit from an integer – it’s more concise and you don’t have to worry about ensuring the mask and shift match.